pubsub/rocketmq/rocketmq.go

```diff
@@ -16,120 +16,247 @@ package rocketmq
 	}
 }
 
 func (r *rocketMQ) Init(metadata pubsub.Metadata) error {
 	var err error
-	r.metadata, err = parseRocketMQMetaData(metadata, r.logger)
+	r.metadata, err = parseRocketMQMetaData(metadata)
 	if err != nil {
 		return err
 	}
+	r.topics = make(map[string]mqc.MessageSelector)
+	r.msgProperties = make(map[string]bool)
+	rlog.SetLogLevel(r.metadata.LogLevel)
+	if r.metadata.MsgProperties != "" {
+		mps := strings.Split(r.metadata.MsgProperties, ",")
+		for _, mp := range mps {
+			r.msgProperties[mp] = true
+		}
+	}
 	r.ctx, r.cancel = context.WithCancel(context.Background())
 
 	return nil
 }
 
 func (r *rocketMQ) setUpConsumer() (mq.PushConsumer, error) {
 	opts := make([]mqc.Option, 0)
-	if r.metadata.ConsumerGroup != "" {
-		opts = append(opts, mqc.WithGroupName(r.metadata.ConsumerGroup))
+	if r.metadata.InstanceName != "" {
+		opts = append(opts, mqc.WithInstance(r.metadata.InstanceName))
 	}
-	if r.metadata.ConsumerBatchSize != 0 {
-		opts = append(opts, mqc.WithPullBatchSize(int32(r.metadata.ConsumerBatchSize)))
+	if r.metadata.ConsumerGroupName != "" {
+		opts = append(opts, mqc.WithGroupName(r.metadata.ConsumerGroupName))
+	}
+	if r.metadata.NameServer != "" {
+		if strings.Contains(r.metadata.NameServer, ",") {
+			opts = append(opts, mqc.WithNameServer(strings.Split(r.metadata.NameServer, ",")))
+		} else if strings.Contains(r.metadata.NameServer, ";") {
+			opts = append(opts, mqc.WithNameServer(strings.Split(r.metadata.NameServer, ";")))
+		} else {
+			opts = append(opts, mqc.WithNameServer([]string{r.metadata.NameServer}))
+		}
 	}
 	if r.metadata.NameSpace != "" {
 		opts = append(opts, mqc.WithNamespace(r.metadata.NameSpace))
 	}
-	if r.metadata.Retries != 0 {
-		opts = append(opts, mqc.WithRetry(r.metadata.Retries))
-	}
 	if r.metadata.NameServerDomain != "" {
 		opts = append(opts, mqc.WithNameServerDomain(r.metadata.NameServerDomain))
 	}
-	if r.metadata.NameServer != "" {
-		opts = append(opts, mqc.WithNameServer(primitive.NamesrvAddr{r.metadata.NameServer}))
-	}
 	if r.metadata.AccessKey != "" && r.metadata.SecretKey != "" {
 		opts = append(opts, mqc.WithCredentials(primitive.Credentials{
-			AccessKey: r.metadata.AccessKey,
-			SecretKey: r.metadata.SecretKey,
+			AccessKey:     r.metadata.AccessKey,
+			SecretKey:     r.metadata.SecretKey,
+			SecurityToken: r.metadata.SecurityToken,
 		}))
 	}
-	return mq.NewPushConsumer(opts...)
+	if r.metadata.Retries > 0 {
+		opts = append(opts, mqc.WithRetry(r.metadata.Retries))
+	}
+	if r.metadata.ConsumerModel != "" {
+		switch strings.ToLower(r.metadata.ConsumerModel) {
+		case "broadcasting":
+			opts = append(opts, mqc.WithConsumerModel(mqc.BroadCasting))
+		case "clustering":
+			opts = append(opts, mqc.WithConsumerModel(mqc.Clustering))
+		default:
+			r.metadata.ConsumerModel = "Clustering"
+			opts = append(opts, mqc.WithConsumerModel(mqc.Clustering))
+			r.logger.Warnf("%s Consumer Model[%s] is error, expected [BroadCasting, Clustering], "+
```

Comment:
`"%s Consumer Model[%s] is invalid: expected [broadcasting, clustering]; we will use default model [clustering]"`

Reply:
ok, thanks